editor/renames_map_3_to_4.cpp

```diff
@@ -154,6 +154,99 @@ const char *RenamesMap3To4::enum_renames[][2] = {
+	{ "JOY_BUTTON_7", "JOY_AXIS_TRIGGER_RIGHT" },
+	{ "JOY_BUTTON_8", "JOY_BUTTON_LEFT_STICK" },
+	{ "JOY_BUTTON_9", "JOY_BUTTON_RIGHT_STICK" },
+	{ "JOY_BUTTON_10", "JOY_BUTTON_BACK" },
+	{ "JOY_BUTTON_11", "JOY_BUTTON_START" },
+	{ "JOY_BUTTON_12", "JOY_BUTTON_DPAD_UP" },
+	{ "JOY_BUTTON_13", "JOY_BUTTON_DPAD_DOWN" },
+	{ "JOY_BUTTON_14", "JOY_BUTTON_DPAD_LEFT" },
+	{ "JOY_BUTTON_15", "JOY_BUTTON_DPAD_RIGHT" },
+	{ "JOY_BUTTON_16", "JOY_BUTTON_GUIDE" },
+	{ "JOY_BUTTON_17", "JOY_BUTTON_MISC1" },
+	{ "JOY_BUTTON_18", "JOY_BUTTON_PADDLE1" },
+	{ "JOY_BUTTON_19", "JOY_BUTTON_PADDLE2" },
+	{ "JOY_BUTTON_20", "JOY_BUTTON_PADDLE3" },
+	{ "JOY_BUTTON_21", "JOY_BUTTON_PADDLE4" },
+	{ "JOY_BUTTON_22", "JOY_BUTTON_TOUCHPAD" },
+	{ "JOY_BUTTON_MAX", "JOY_BUTTON_MAX" },
+	{ "JOY_DPAD_DOWN", "JOY_BUTTON_DPAD_DOWN" },
+	{ "JOY_DPAD_LEFT", "JOY_BUTTON_DPAD_LEFT" },
+	{ "JOY_DPAD_RIGHT", "JOY_BUTTON_DPAD_RIGHT" },
+	{ "JOY_DPAD_UP", "JOY_BUTTON_DPAD_UP" },
+	{ "JOY_SELECT", "JOY_BUTTON_BACK" },
+	{ "JOY_GUIDE", "JOY_BUTTON_GUIDE" },
+	{ "JOY_START", "JOY_BUTTON_START" },
+	{ "JOY_MISC1", "JOY_BUTTON_MISC1" },
+	{ "JOY_PADDLE1", "JOY_BUTTON_PADDLE1" },
+	{ "JOY_PADDLE2", "JOY_BUTTON_PADDLE2" },
+	{ "JOY_PADDLE3", "JOY_BUTTON_PADDLE3" },
+	{ "JOY_PADDLE4", "JOY_BUTTON_PADDLE4" },
+	{ "JOY_L", "JOY_BUTTON_LEFT_SHOULDER" },
+	{ "JOY_R", "JOY_BUTTON_RIGHT_SHOULDER" },
+	{ "JOY_L3", "JOY_BUTTON_LEFT_STICK" },
+	{ "JOY_R3", "JOY_BUTTON_RIGHT_STICK" },
+	{ "JOY_TOUCHPAD", "JOY_BUTTON_TOUCHPAD" },
+	{ "JOY_XBOX_A", "JOY_BUTTON_A" },
+	{ "JOY_XBOX_B", "JOY_BUTTON_B" },
+	{ "JOY_XBOX_X", "JOY_BUTTON_X" },
+	{ "JOY_XBOX_Y", "JOY_BUTTON_Y" },
+	{ "JOY_DS_A", "JOY_BUTTON_B" },
+	{ "JOY_DS_B", "JOY_BUTTON_A" },
+	{ "JOY_DS_X", "JOY_BUTTON_Y" },
+	{ "JOY_DS_Y", "JOY_BUTTON_X" },
+	{ "JOY_SONY_CIRCLE", "JOY_BUTTON_B" },
+	{ "JOY_SONY_SQUARE", "JOY_BUTTON_X" },
+	{ "JOY_SONY_TRIANGLE", "JOY_BUTTON_Y" },
+	{ "JOY_SONY_X", "JOY_BUTTON_A" },
+	{ "JOY_WII_C", "JOY_BUTTON_RIGHT_SHOULDER" },
+	{ "JOY_WII_Z", "JOY_AXIS_TRIGGER_LEFT" },
+	{ "JOY_WII_MINUS", "JOY_BUTTON_BACK" },
+	{ "JOY_WII_PLUS", "JOY_BUTTON_START" },
+	{ "JOY_OCULUS_AX", "JOY_AXIS_TRIGGER_RIGHT" },
+	{ "JOY_OCULUS_BY", "JOY_BUTTON_B" },
+	{ "JOY_OCULUS_MENU", "JOY_BUTTON_Y" },
+	{ "JOY_VR_PAD", "JOY_BUTTON_DPAD_LEFT" },
+	{ "JOY_VR_TRIGGER", "JOY_BUTTON_DPAD_RIGHT" },
+	{ "JOY_VR_GRIP", "JOY_BUTTON_X" },
+	{ "JOY_VR_ANALOG_GRIP", "JOY_BUTTON_LEFT_SHOULDER" },
+	{ "JOY_VR_ANALOG_TRIGGER", "JOY_BUTTON_X" },
+	{ "JOY_OPENVR_MENU", "JOY_BUTTON_B" },
+
+	// Axes
+	{ "JOY_ANALOG_LX", "JOY_AXIS_LEFT_X" },
+	{ "JOY_ANALOG_LY", "JOY_AXIS_LEFT_Y" },
+	{ "JOY_ANALOG_RX", "JOY_AXIS_RIGHT_X" },
+	{ "JOY_ANALOG_RY", "JOY_AXIS_RIGHT_Y" },
+	{ "JOY_ANALOG_L2", "JOY_AXIS_TRIGGER_LEFT" },
+	{ "JOY_ANALOG_R2", "JOY_AXIS_TRIGGER_RIGHT" },
+	{ "JOY_AXIS_0", "JOY_AXIS_LEFT_X" },
+	{ "JOY_AXIS_1", "JOY_AXIS_LEFT_Y" },
+	{ "JOY_AXIS_2", "JOY_AXIS_RIGHT_X" },
+	{ "JOY_AXIS_3", "JOY_AXIS_RIGHT_Y" },
+	{ "JOY_AXIS_4", "null" },
```

Comment:
Is there no mapping for `JOY_AXIS_4` ?

Reply:
Godot 3's docs describe it as "Generic gamepad axis 4.", for which there's no equivalency in the godot 4 enums. For the few cases like that I thought mapping them to `null` would be safer than mapping them to any other value which may not be the one originally intended by the user.